src/conn/conn_compact.c

```diff
@@ -465,6 +465,14 @@ __background_compact_server(void *arg)
              * have been parsed.
              */
             if (uri->size == 0 || full_iteration) {
+                /* If the server was configured to run once, after a full iteration, we are done. */
+                if (full_iteration && conn->background_compact.run_once) {
```

Comment:
I am not a fan of that specific check, I wish there was a better way to change the background compaction state and update the relevant stat in one place. @Sean04, who do you think we could improve this?